meshnets/data_processing/data_statistics.py

```diff
@@ -0,0 +1,79 @@
+"""File with functions to compute data statistics"""
+import numpy as np
+
+
+class OnlineStatistics:
+    """Computes mean ad standard deviation from streaming data
+
+    Args:
+      axis: Axis over which to compute. Default is None, which means
+        that the statistics are computed over the flattened array.
+
+    Example:
+
+    >>> dataset = [[1, 2], [3, 4]]
+    >>> stats = OnlineStatistics()
+    >>> for data in dataset:
+    ...     stats.update(data)
+    >>> stats.get_mean()
+    array([[2.5]])
+    >>> stats.get_std()
+    array([[1.11803399]])
+    >>> stats = OnlineStatistics(axis=0)
+    >>> for data in dataset:
+    ...     stats.update(data)
+    >>> stats.get_mean()
+    array([[2., 3.]])
+    >>> stats.get_std()
+    array([[1., 1.]])
+    >>> stats = OnlineStatistics(axis=1)
+    >>> for data in dataset:
+    ...     stats.update(data)
+    >>> stats.get_mean()
+    array([[1.5], [3.5]])
+    >>> stats.get_std()
+    array([[0.5], [0.5]])
+    """
+
+    def __init__(self, axis=None):
+        self.axis = axis
+
+        self.n = None
+        self.cumulative_sum = None
+
+        self.cumulative_sum_2 = None
+
+    def update(self, example):
+        data_point = np.expand_dims(example, 0)
+        self.sum_element = np.sum(data_point, axis=self.axis, keepdims=True)
+        self.sum_element_2 = np.sum(data_point**2,
+                                    axis=self.axis,
+                                    keepdims=True)
+        if self.cumulative_sum is None:
+            self.cumulative_sum = self.sum_element
+            self.cumulative_sum_2 = self.sum_element_2
+        else:
+            sum_element = np.sum(data_point, axis=self.axis, keepdims=True)
```

Comment:
This `sum_element` and `sum_element_2` seem to be exactly the same code you present above with the `self`. So this isn't really required to have here again to thereafter compute the cumulative sums.

Reply:
Completely agree. Must have missed that when addressing Hugo's comment